staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -343,6 +370,9 @@ func (c *dynamicResourceClient) Apply(ctx context.Context, name string, obj *uns
 	}
 	patchOpts := opts.ToPatchOptions()
 
+	if err := c.validateURLSegments(name); err != nil {
```

Comment:
Personally I would have validated the name immediately after checking the length, rather than doing so in the middle of the function here, to avoid doing the more expensive operations/allocations. 

Looking around at the other functions, it seems pretty inconsistent whether the validation check was placed early in the function body vs. immediately before the HTTP request. I think at least it should be uniform

Reply:
I've always called the `validateURLSegments` method right before calling the `makeURLSegments`, but I can move things around as you mentioned.

Reply:
Yeah, I think it was consistent, but I agree that making it early is probably better (moved a lot before running encoding of the object). Moved.